Sources/Turf/GeoJSON.swift

```diff
@@ -0,0 +1,358 @@
+import Foundation
+#if !os(Linux)
+import CoreLocation
+#endif
+
+
+public class BaseFeature: Codable {
+    private enum CodingKeys: String, CodingKey {
+        case properties
+    }
+    
+    public var properties: [String : AnyJSONType]?
+    
+    public required init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        properties = try container.decode([String : AnyJSONType]?.self, forKey: .properties)
+    }
+    
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encode(properties, forKey: .properties)
+    }
+}
+
+public class Feature: BaseFeature {
+    public var type: GeoJSONType
+    
+    // Used to extract the geometry’s type w/o double decoding its coordinates
+    fileprivate var simplifiedGeometry: Geometry?
+    
+    private enum CodingKeys: String, CodingKey {
+        case type
+        //case properties
+        case simplifiedGeometry = "geometry"
+    }
+    
+    public required init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        simplifiedGeometry = try container.decodeIfPresent(Geometry.self, forKey: .simplifiedGeometry)
+        type = try GeoJSONType(rawValue: container.decode(String.self, forKey: .type))!
+        
+        try super.init(from: decoder)
+    }
+    
+    public override func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encode(type.rawValue, forKey: .type)
+        try super.encode(to: encoder)
+    }
+}
+
+public struct Geometry: Codable {
+    public var type: String
+}
+
+/**
+ A `LineString` struct represents a shape consisting of two or more coordinates,
+ specified as `[CLLocationCoordinate2D]`
+ */
+public struct LineString: Codable {
+    var type: String = GeometryType.LineString.rawValue
+    var coordinates: [CLLocationCoordinate2D]
+}
+
+public struct Polygon: Codable {
+    var type: String = GeometryType.Polygon.rawValue
+    var coordinates: [[CLLocationCoordinate2D]]
+    
+    init(coordinates: [[CLLocationCoordinate2D]]) {
+        self.coordinates = coordinates
+    }
+    
+    var innerRings: [Ring]? {
+        get { return Array(coordinates.suffix(from: 1)).map { Ring(coordinates: $0) } }
+    }
+    
+    var outerRing: Ring? {
```

Comment:
A polygon without an outer ring is degenerate. Is there anything we can do to make this property non-optional?